core/src/main/java/feign/codec/ErrorDecoder.java

```diff
@@ -76,13 +73,15 @@
    *                  ex. {@code IAM#getUser()}
    * @param response  HTTP response where {@link Response#status() status} is greater than or equal
    *                  to {@code 300}.
-   * @return Exception IOException, if there was a network error reading the response or an
-   * application-specific exception decoded by the implementation. If the throwable is retryable, it
-   * should be wrapped, or a subtype of {@link RetryableException}
+   * @return Exception An Exception or <code>null</code>. IOException, if there was a network error
+   * reading the response or an application-specific exception decoded by the implementation. If
+   * the throwable is retryable, it should be wrapped, or a subtype of {@link RetryableException}.
+   * If <code>null</code> is returned no exception is thrown and the normal {@link Decoder} will
+   * be asked to decode the response.
    */
-  public Exception decode(String methodKey, Response response);
+  Exception decode(String methodKey, Response response);
```

Comment:
Interface methods are public by default. The 'public' is redundant here. The IDE flagged this.

I will re-add it if the code style guide requires it.